doc/contributing/frontend/assets.rst

```diff
@@ -0,0 +1,152 @@
+======
+Assets
+======
+
+.. Note:: Assets are only supported in the CKAN 2.9 and above. If you
+          are using CKAN <= 2.8, refer to legacy `Fanstatic resources
+          <https://docs.ckan.org/en/2.8/contributing/frontend/resources.html>`_
+          documentation.
+
+Assets are .css and .js files that may be included in an html page.
+Assets are included in the page by using the ``{% asset %}`` tag and
+CKAN uses `Webassets <https://webassets.readthedocs.io/en/latest/>`_
+to serve these assets.
+
+::
+
+ {% asset 'library_name/asset_name' %}
+
+Assets are grouped into libraries and the full asset name consists of
+``<library name>/<asset name>``. For example:
+
+::
+
+ {% asset 'my_webassets_library/my_javascript_file.js' %}
+
+It is important to note that these assets will be added to the page as
+defined by the assets, not in the location of the ``{% asset %}`` tag.
+Duplicate assets will not be added and any dependencies will be included as
+well as the assets, all in the correct order (see below for details).
+
+Libraries can be added to CKAN from extensions using a helper function
+within the toolkit. See below.
+
+In debug mode assets are served un-minified and unbundled (each asset is
+served separately). In non-debug mode the files are served minified and bundled
+(where allowed).
+
+.. Note::
+    .js and .css assets should be supplied as un-minified files.  Minified
```

Comment:
@smotornyuk Two questions here:

1. Do you mean that when commiting files to this repo or an extension one you should not commit the minified ones?
2. With webassets do we still need to build the minified files with `ckan minify` or whatever before a release?

Reply:
1. Exactly - you need to commit only original source, without any preprocessing.
2. Generally, no. All JS files with `rjsmin` are minified automatically when `debug = false` (example: https://github.com/ckan/ckan/blob/master/ckan/public/base/javascript/webassets.yml#L2) As for css, right now styles are not minified as there is no built-in/default css-minifier provided by webassets. I think about including one additional dependency, so that styles will be handled in the same way as scripts, but i want to check first, whether there is some option that doesn't require updating dependencies